x-pack/plugin/esql/src/test/java/org/elasticsearch/xpack/esql/action/EsqlQueryResponseTests.java

```diff
@@ -368,6 +382,58 @@ public void testBasicXContentIdAndRunning() {
         }
     }
 
+    public void testNullColumnsXContentDropNulls() {
+        try (
+            EsqlQueryResponse response = new EsqlQueryResponse(
+                List.of(new ColumnInfo("foo", "integer"), new ColumnInfo("all_null", "integer")),
+                List.of(new Page(blockFactory.newIntArrayVector(new int[] { 40, 80 }, 2).asBlock(), blockFactory.newConstantNullBlock(2))),
+                null,
+                false,
+                null,
+                false,
+                false
+            )
+        ) {
+            assertThat(
+                Strings.toString(wrapAsToXContent(response), new ToXContent.MapParams(Map.of(DROP_NULL_COLUMNS_OPTION, "true"))),
+                equalTo("{" + """
+                    "null_columns":[{"name":"all_null","type":"integer"}],""" + """
+                    "columns":[{"name":"foo","type":"integer"}],""" + """
+                    "values":[[40],[80]]}""")
+            );
```

Comment:
Small nit - we might want to preserve the order of the columns even when removing null columns. 
One way to do that is to list the columns as in the regular response but indicate it is null. we can then either skip or add a null in place:
```json
"columns":[{"name":"foo","type":"integer"}, {"name":"all_null", "type":"integer", "null" : true}]
"values":[[40],[80]]
```
or
```json 
"values":[[40, null],[80, null]]
```

Reply:
> I'm not sure that we wanna go there, since the task at hand is clear - "drop null columns", but a more generalise view of this is "constant_columns" - a column with a constant value, which in this case would be null.

Oh that's kind of neat! It's quite easy for us to detect constant columns too.

> Small nit - we might want to preserve the order of the columns even when removing null columns.

I like the idea. Then I don't like the idea. Then I do. I dunno. I kind of like that, no matter what, the way to parse the columns is by looking at the `columns` header. You just one for one them. No need for checking. As it stands now if you send this parameter the null columns just "disappear" from the parsing logic. They are only sitting off to the side as metadata. Like, no need to have different parsing code to handle this case.

OTOH, Chris' quite neat proposal around constant columns would be much more compatible with keeping one list of columns. Just sometimes having a constant. I suppose @stratoula could do either one. I wonder what our friends who work on the clients think. I'll ask!

Reply:
One interesting thing about extracting constant values is that it's something that would likely also come as part of supporting some different output format like parquet or arrow flight or whatever. I presume those formats have specifications for, well, the entire response shape. And we'd do whatever makes sense for them.

Reply:
Note also that our client developer friends asked for an "object" flavor of the response in #103263 - it'd be opt in. And if you did opt in the `values` response would be the same:
```
  "values": [
    {"r1": 1, "r2": 2},
    {"r1": 3, "r2": 4}
  ]
```

Of special note - the "object" flavor of the response is not compatible with a `columnar` response. That's not the default option for us, but it *is* the fastest way we can return the values so I'd hope the official clients could use it where possible.

Reply:
From the kibana side, I am happy to work with whatever response makes sense for you. I just need an indicator!

Reply:
Thr response should be consistent as much as possible regardless of the options used. Generalizing the constant just like skipping null is as compression of sorts that's better handled separately (columnar option, transparent zip or cobra or the other formats we're looking at).

Reply:
```
{
  "columns": [
    {"name": "n1", "type": "long", "constant_value": null},
    {"name": "r1", "type": "long"},
    {"name": "n2", "type": "keyword", "constant_value": null},
    {"name": "r2", "type": "long"},
  },
  "values": [
    [1, 2],
    [3, 4]
  ]
}
```
On one hand this format seems better because it keeps the order of columns as the user asks for it, which otherwise would be lost if we group the null/constant value columns grouped under a separate field, as in the alternative, 1st example.
OTOH, the `values` not being in sync with the `columns` elements-count-wise looks a bit off. But it's a worthy tradeoff, IMO.

Also, this way we can rename the URL parameter - `drop_null_columns` - which IMO is a misnomer, as it only separates the columns.

Reply:
> Generalizing the constant just like skipping null is as compression of sorts that's better handled separately (columnar option, transparent zip or cobra or the other formats we're looking at).

I *kind* of buy this, but we're in a place where we can know a lot more than a compression algorithm. OTOH, a different output structure like Apache arrow would integrate with this just fine. They have a "null block" equivalent.

Reply:
FTR, I've discussed this with Nik and come with the following conclusion:
columns and values should be kept in sync to avoid breaking existing parsing for consumers that set the option and don't care about the null values
for those that do, to maintain order, the list of all columns (null and not null) is returned as a separate section.
So something like:

```json
{
  "all_columns": [
    {"name": "n1", "type": "long"},
    {"name": "r1", "type": "long"},
    {"name": "n2", "type": "keyword"}
    {"name": "r2", "type": "long"},
  ],
  "columns": [
    {"name": "r1", "type": "long"},
    {"name": "r2", "type": "long"}
  ],
  "values": [
    [1, 2],
    [3, 4]
  ]
}
```

This way clients that do care about the null columns or the available info have the data available.

Reply:
So in order to find the columns with nulls you need to find the difference between all_columns and columns? Do I get this correctly?

Reply:
Yeah. The advantage of this way is you get everything back in the right order. I figured *someone* would want that. We talked about putting a `always_null` on the column in `all_columns` so you could do it without the set difference operator. I decided to do it if someone asked for it but not at first.

Reply:
I am fine with that 👍

Reply:
👍